docs/communicate-with-javascript.md

```diff
@@ -1669,3 +1670,485 @@ nullable]` above. Currently 4 directives are supported: `null_to_opt`,
+generated:
+
+```ocaml
+val actionToJs : action -> int
+
+val actionFromJs : int -> action option
+```
+
+The `{ jsConverter = newType }` payload can also be passed to `@bs.deriving`
+attribute when working with polymorphic variants.
+
+### Records
+
+#### Accessing fields
+
+Use `@bs.deriving accessors` on a record type to create accessors for its record
+field names.
+
+```ocaml
+type pet = { name : string } [@@bs.deriving accessors]
+
+let pets = [| { name = "Brutus" }; { name = "Mochi" } |]
+
+let () = pets |. Belt.Array.map name |. Js.Array2.joinWith "&" |. Js.log
+```
+
+Melange will generate a function for each field defined in the record. In this
+case, a function `name` that allows to get that field from any record of type
+`pet`:
+
+```ocaml
+let name (param : pet) = param.name
+```
+
+Considering all the above, the produced JavaScript will be:
+
+```js
+function name(param) {
+  return param.name;
+}
+
+var pets = [
+  {
+    name: "Brutus"
+  },
+  {
+    name: "Mochi"
+  }
+];
+
+console.log(Belt_Array.map(pets, name).join("&"));
+```
+
+#### Convert records into abstract types
+
+Usually, it’s recommended to use plain records when compiling to JavaScript
+objects (see the section on [binding to JavaScript
+objects](#bind-to-javascript-objects)). But there's a specific case where
+records may not be enough. This happens when a record type has optional fields,
+and we want the resulting JavaScript objects to exclude those fields entirely.
+
+For instance, consider the following record:
+
+```ocaml
+type person = {
+  name : string;
+  age : int option;
+}
+```
+
+The specific use-case we are referring to arises when we have a Melange value
+such as `{ name = "John"; age = None }`, but we desire the resulting JavaScript
+object to be `{name: "Carl"}`.
+
+To address this case, the `@bs.deriving abstract` attribute can prove useful.
+When applied to a record type, this attribute removes the record type definition
+and replaces it with an abstract type. Additionally, it generates these
+functions:
+
+- A constructor function for creating values of the type
+- Getters and setters for accessing the record fields
+
+Let’s see an example. Considering this Melange code:
+
+```ocaml
+type person = {
+  name : string;
+  age : int; [@bs.optional]
+}
+[@@bs.deriving abstract]
+```
+
+Melange will transform the `person` type into an abstract type and generate the
+constructor, getters, and setters functions. So we would get an interface like
+this:
+
+```ocaml
+type person
+
+val person : name:string -> ?age:int -> unit -> person
+
+val nameGet : person -> string
+
+val ageGet : person -> int option
+```
+
+The `person` constructor enables the creation of values for the `person` type.
+It is the only valid way to create such values because using literals directly
+like `{ name = "Alice"; age = None }` will no longer type check, due to the
+opaque nature of the abstract type definition.
```

Comment:
```suggestion
The `person` function can be used to create values of `person`. It is the only
possible way to create values of this type, since Melange makes it abstract.
Using literals like `{ name = "Alice"; age = None }` directly doesn't type check.
```